lib/iris/tests/unit/lazy_data/test_array_masked_to_nans.py

```diff
@@ -62,6 +68,14 @@ def test_empty_mask(self):
         self._common_checks(result)
         self.assertArrayAllClose(result, masked_array.data)
 
+    def test_no_mask(self):
+        masked_array = ma.masked_array([1.0, 2.0], mask=ma.nomask)
+
+        result = array_masked_to_nans(masked_array)
+
+        self._common_checks(result)
+        self.assertArrayAllClose(result, masked_array.data)
+
     def test_integer_array(self):
```

Comment:
I think it is worth renaming this `test_masked_ints` to make clear the relationship to `test_unmasked_ints`.
Also, as this is the one that changes the type, it is worth checking the actual values, as in `test_masked_input`.